src/azure-cli-core/azure/cli/core/commands/parameters.py

```diff
@@ -26,6 +26,12 @@ def get_location_completion_list(prefix, **kwargs):#pylint: disable=unused-argum
     result = get_subscription_locations()
     return [l.name for l in result]
 
+def location_name_type(name):
+    if ' ' in name:
+        # if display name is provided, attempt to convert to short form name
+        name = next((l.name for l in get_subscription_locations() if l.display_name == name), name)
```

Comment:
for this case, it will use the original name, which is fine mostly.
I didn't call this out again since we thought we will address that through a different PR. But i agree we can do case insensitive check right now since it is cheap.